cocotb/utils.py

```diff
@@ -35,12 +35,12 @@
 import functools
 import warnings
 
-if "COCOTB_SIM" in os.environ:
-    from cocotb import simulator
-    _LOG_SIM_PRECISION = simulator.get_precision()  # request once and cache
-else:
-    simulator = None
-    _LOG_SIM_PRECISION = -15
+from cocotb import simulator
+
+
+@functools.lru_cache(maxsize=1)
+def _get_simulator_precision():
+    return simulator.get_precision()
```

Comment:
One final option:
```suggestion
def _get_simulator_precision():
    # cache and replace this function
    _get_simulator_precision.__code__ = (lambda: simulator.get_precision()).__code__
    return _get_simulator_precision()
```

Possibly a micro-optimization, but I'd expect it to be significantly faster.

Reply:
That's 56ns, I'll put that it.

Reply:
That's a sizeable improvement, great!